### PR TITLE
[RUM-4479] Update Android & iOS SDKs

### PR DESCRIPTION
**Pull request by marco-saia-datadog**
### What does this PR do?

Updates Android SDK to 2.10.1 and iOS SDK to 2.13.0.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: packages/core/android/build.gradle [hunk 1]
```diff
@@ -188,10 +188,10 @@ dependencies {
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
   compileOnly "com.squareup.okhttp3:okhttp:3.12.13"
   
-  implementation "com.datadoghq:dd-sdk-android-rum:2.6.2"
-  implementation "com.datadoghq:dd-sdk-android-logs:2.6.2"
-  implementation "com.datadoghq:dd-sdk-android-trace:2.6.2"
-  implementation "com.datadoghq:dd-sdk-android-webview:2.6.2"
+  implementation "com.datadoghq:dd-sdk-android-rum:2.10.1"
```
Comment (0xnm) [inline packages/core/android/build.gradle]: Android SDK already has a version 2.11.0 https://github.com/DataDog/dd-sdk-android/releases/tag/2.11.0, it brings mostly tracing code update + Otel + Session Replay updates. Do you want to use this version in this PR instead?

File: packages/react-native-session-replay/android/build.gradle [hunk 2]
```diff
@@ -188,7 +188,7 @@ dependencies {
     api "com.facebook.react:react-android:$reactNativeVersion"
   }
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-  implementation "com.datadoghq:dd-sdk-android-session-replay:2.6.2"
+  implementation "com.datadoghq:dd-sdk-android-session-replay:2.10.1"
```
Comment (0xnm) [inline packages/react-native-session-replay/android/build.gradle]: same: maybe 2.11.0?

**Review by 0xnm — APPROVED** [review]
nice! the only suggestion is to use Android SDK 2.11.0

File: packages/core/android/build.gradle (hunk 1 above)
Comment (marco-saia-datadog) [reply packages/core/android/build.gradle]: You are right. I will give it a try, and if there are no major blockers we can use it.

**Pull request merged**